### PR TITLE
support per device/group port scan

### PR DESCRIPTION
**Pull request by wzjjack**

### CODE REVIEW

File: sensor/DeviceServiceScanSensor.js [hunk 1]
```diff
@@ -15,138 +15,250 @@
...
-      };
-    } catch(err) {
-      log.error('Failed to scan: ', err);
-    };
+    async applyPolicy(host, ip, policy) {
+        log.info("Applying device service scan policy:", ip, policy);
+        try {
+            if (ip === '0.0.0.0') {
+                if (policy === true) {
+                    this.systemSwitch = true;
```
Comment (drahnieR) [inline sensor/DeviceServiceScanSensor.js:61]: looks like `systemSwitch` and `adminSystemSwitch` are the same thing.
policy:system all targets 0.0.0.0

File: sensor/DeviceServiceScanSensor.js [hunk 2]
```diff
@@ -15,138 +15,250 @@
...
+                }
+            }
+        } catch (err) {
+            log.error("Got error when applying adblock policy", err);
+        }
+    }
 
-  async runOnce() {
-    log.info('Scan start...');
+    async applyScan() {
```
Comment (drahnieR) [inline sensor/DeviceServiceScanSensor.js:143]: this is a little bit too complex form what i can see.

Can we have a global list of all devices, and mark them as if them should be scanned on each iteration, by system -> network -> group -> device policy? Device policy comes at last and have the highest priority.

And there should be only one interval timer to control the scan. which is controlled by the global switch.

File: sensor/DeviceServiceScanSensor.js (hunk 1 above)
Comment (wzjjack) [reply sensor/DeviceServiceScanSensor.js:61]: Nope, they aer different, adminSystemSwitch means feature on or off,  systemSwitch means target=>0.0.0.0 policy is true or  false.

File: sensor/DeviceServiceScanSensor.js (hunk 2 above)
Comment (wzjjack) [reply sensor/DeviceServiceScanSensor.js:143]: I thought about this way, but the policy may be async, we need to control them by different level.

Comment (wzjjack) [reply sensor/DeviceServiceScanSensor.js:143]: e.g. 
if the device1 port scan is disabled, but tag1 port scan is enabled. will clear device1 scan timer and create a scan job for tag1, tag1 may include device1. when box restart, we may not confirm which policy will be applied first...

Comment (drahnieR) [reply sensor/DeviceServiceScanSensor.js:143]: go through every policy on each iteration, port scan is a heavier job than reading redis. Is it even possible in the current design to exclude a device from scanning in a network?

File: sensor/DeviceServiceScanSensor.js (hunk 1 above)
Comment (drahnieR) [reply sensor/DeviceServiceScanSensor.js:61]: consider using `Config.isFeatureOn()` maybe? this is confusing

Comment (drahnieR) [reply sensor/DeviceServiceScanSensor.js:61]: consider using `Config.isFeatureOn()` maybe? this is confusing

Comment (wzjjack) [reply sensor/DeviceServiceScanSensor.js:61]: `Config.isFeatureOn()` is fine, but this sensor inherits `senor`, will call `hookFeature` which may listenning feature on and off, so we apply a new variable to handle this. Just keep the structure currently.

**Pull request closed without merging**